rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/TechnologyIdentified.java

```diff
@@ -0,0 +1,146 @@
+            implements TechnologyIdentifiedWithCount, TechnologyIdentifiedWithName
+{
+    public static final int DEFAULT_COUNT = 1;
+
+    private static Logger LOG = Logging.get(TechnologyIdentified.class);
+
+    private String technologyName;
+    private Set<String> tags = new HashSet<>();
+    private int count = DEFAULT_COUNT;
+
+    /**
+     * Creates the operation with the given technology name.
+     */
+    public TechnologyIdentified(String technologyName)
+    {
+        this.technologyName = technologyName;
+    }
+
+    /**
+     * Creates the operation with the given input variable and technology name.
+     */
+    public TechnologyIdentified(String variableName, String technologyName)
+    {
+        super(variableName);
+        this.technologyName = technologyName;
+    }
+
+    /**
+     * Create the operation with the given technology name.
+     */
+    public static TechnologyIdentifiedWithName named(String technologyName)
+    {
+        return new TechnologyIdentified(technologyName);
+    }
+
+    /**
+     * Sets the number of items that have been found by this operation.
+     *
+     * This is optional, and if left unspecified, it will default to {@link TechnologyIdentified#DEFAULT_COUNT}.
+     */
+    public TechnologyIdentifiedWithCount numberFound(int count)
+    {
+        this.count = count;
+        return this;
+    }
+
+    /**
+     * Specifies a tag to associate with this technology.
+     */
+    public TechnologyIdentified withTag(String tag)
+    {
+        this.tags.add(tag);
+        return this;
+    }
+
+    /**
+     * Contains the name of the technology identified by this rule.
+     */
+    public String getTechnologyName()
+    {
+        return technologyName;
+    }
+
+    /**
+     * Contains the set of tags identified associated with this technology.
+     */
+    public Set<String> getTags()
+    {
+        return tags;
+    }
+
+    /**
+     * Contains the count of items located by this operation (usually '1', and this is the default if nothing is specified).
+     */
+    public int getCount()
+    {
+        return count;
+    }
+
+    @Override
+    public void perform(GraphRewrite event, EvaluationContext context, WindupVertexFrame payload)
+    {
+        ProjectModel project;
+        if (payload instanceof ProjectModel)
+        {
+            project = (ProjectModel) payload;
+        }
+        else if (payload instanceof FileModel)
+        {
+            project = ((FileModel) payload).getProjectModel();
+        } else if (payload instanceof FileLocationModel)
+        {
+            project = ((FileLocationModel)payload).getFile().getProjectModel();
+        }
+        else
+        {
+            return;
+        }
+
+        TechnologyUsageStatisticsService service = new TechnologyUsageStatisticsService(event.getGraphContext());
+        TechnologyUsageStatisticsModel model = service.get(project, this.technologyName);
```

Comment:
Maybe rename `get` method to `getOrCreate` (as other methods below), since it creates new `TechnologyUsageStatisticsModel` if it doesn't exist.

Reply:
+1 - I have made this change. I hate a method called "get" that actually creates, and yet I've managed to make that mistake myself. Oops.

Good catch! :)